semanticdb/semanticdb3/semanticdb3.md

```diff
@@ -953,6 +958,122 @@ diagnostics as error, warning, information or hint.
   </tr>
 </table>
 
+### Synthetic 
+
+```protobuf
+message Synthetic {
+  Range range = 1;
+  Tree tree = 2;
+}
+```
+
+"Synthetics" is a section of a [TextDocument](#textdocument) that stores
+synthetic terms added by compilers that do not appear in the original source
+file. If synthetics appear in a `TextDocument`,  then the `TextDocument` should
+also have a `text` field.
```

Comment:
I don't think it makes much sense to have a range when there's nothing available that it refers to. But I took this out, because I suppose it is unnecessarily strict.

Reply:
SemanticDB payloads are not supposed to be self-sufficient. If `text` is empty, we can still retain a reference to the original source file via `uri`. Occurrences are in the same situation.

Reply:
Thank you for clarifying, that makes sense.